docs/specs/m2m.md

```diff
@@ -320,97 +320,99 @@ In essence, Auth0 offers
 - APIs are still required to handle `sub` conditionally, by checking whether `sub` ends with `@clients` or not.
 - Work around the constraint that `audience` is once and only once, by allowing the developer to exchange token with another `audience`.
 
-If we were to follow the same, we need to model Resource and Scope as database tables, and introduce relationship between:
+To adopt a similar functionality, I propose:
 
-- User and scope (direct association)
-- Role and scope
-- There is no relationship between Group and scope, as a Group is a collection of Roles.
+- Model Resource and Scope as database tables, so that Role and User can have relationship with Scope.
+- Allow relationship between User and Scope (direct association).
+- Allow relationship between Role and Scope.
+- No relationship between Group and Scope to keep things simple. After all, a Group is just a collection of Roles.
 
-### Discussion: (Future works) About the handling of `access_token` in Resources
+In contrast to Auth0, `resource` **IS NOT** allowed when `grant_type=authorization_code`.
+By disallowing `resource` in `grant_type=authorization_code`,
+the `aud` in the issued `access_token` is always a singleton array containing `https://auth.myapp.com`.
+Thus, `scope` always mean the access of `client_id` to `aud` acting on behalf of `sub`.
 
-> [!NOTE]
-> This section is only relevant when we allow the developer to associate scope to User and role.
+With this constraint, the `access_token` of `grant_type=authorization_code` always look like:
 
-The pain-points are
+```json
+{
+  "iss": "https://auth.myapp.com",
+  "sub": "johndoe",
+  "aud": ["https://auth.myapp.com"],
+  "client_id": "mobileapp",
+  "roles": ["onlinestore:admin"],
```

Comment:
The current key of roles is `https://authgear.com/claims/user/roles`.

Reply:
Thanks. Fixed.